test-data/unit/check-incremental.test

```diff
@@ -4362,6 +4362,297 @@ import b
+
+A(1) < A(2)
+A(1) <= A(2)
+A(1) > A(2)
+A(1) >= A(2)
+A(1) == A(2)
+A(1) != A(2)
+
+A(1) < 1  # E: Unsupported operand types for < ("A" and "int")
+A(1) <= 1  # E: Unsupported operand types for <= ("A" and "int")
+A(1) > 1  # E: Unsupported operand types for > ("A" and "int")
+A(1) >= 1  # E: Unsupported operand types for >= ("A" and "int")
+A(1) == 1
+A(1) != 1
+
+1 < A(1)  # E: Unsupported operand types for > ("A" and "int")
+1 <= A(1)  # E: Unsupported operand types for >= ("A" and "int")
+1 > A(1)  # E: Unsupported operand types for < ("A" and "int")
+1 >= A(1)  # E: Unsupported operand types for <= ("A" and "int")
+1 == A(1)
+1 != A(1)
+
+[file a.py]
+from dataclasses import dataclass
+
+@dataclass(order=True)
+class A:
+    a: int
+
+[builtins fixtures/attr.pyi]
+[rechecked]
+[stale]
+[out2]
+main:2: error: Revealed type is 'def (a: builtins.int) -> a.A'
+main:3: error: Revealed type is 'def (builtins.object, builtins.object) -> builtins.bool'
+main:4: error: Revealed type is 'def (builtins.object, builtins.object) -> builtins.bool'
+main:5: error: Revealed type is 'def [T] (self: T`1, other: T`1) -> builtins.bool'
+main:6: error: Revealed type is 'def [T] (self: T`1, other: T`1) -> builtins.bool'
+main:7: error: Revealed type is 'def [T] (self: T`1, other: T`1) -> builtins.bool'
+main:8: error: Revealed type is 'def [T] (self: T`1, other: T`1) -> builtins.bool'
+main:17: error: Unsupported operand types for < ("A" and "int")
+main:18: error: Unsupported operand types for <= ("A" and "int")
+main:19: error: Unsupported operand types for > ("A" and "int")
+main:20: error: Unsupported operand types for >= ("A" and "int")
+main:24: error: Unsupported operand types for > ("A" and "int")
+main:25: error: Unsupported operand types for >= ("A" and "int")
+main:26: error: Unsupported operand types for < ("A" and "int")
+main:27: error: Unsupported operand types for <= ("A" and "int")
+
+[case testIncrementalDataclassesSubclassModified]
+from b import B
+B(5, 'foo')
+
+[file a.py]
+from dataclasses import dataclass
+
+@dataclass
+class A:
+    x: int
+
+[file b.py]
+from a import A
+from dataclasses import dataclass
+
+@dataclass
+class B(A):
+    y: str
+
+[file b.py.2]
+from a import A
+from dataclasses import dataclass
+
+@dataclass
+class B(A):
+    y: int
+
+[builtins fixtures/list.pyi]
+[out1]
+[out2]
+main:2: error: Argument 2 to "B" has incompatible type "str"; expected "int"
+[rechecked b]
+
+[case testIncrementalDataclassesSubclassModifiedErrorFirst]
+from b import B
+B(5, 'foo')
+
+[file a.py]
+from dataclasses import dataclass
+
+@dataclass
+class A:
+    x: int
+
+[file b.py]
+from a import A
+from dataclasses import dataclass
+
+@dataclass
+class B(A):
+    y: int
+
+[file b.py.2]
+from a import A
+from dataclasses import dataclass
+
+@dataclass
+class B(A):
+    y: str
+
+[builtins fixtures/list.pyi]
+[out1]
+main:2: error: Argument 2 to "B" has incompatible type "str"; expected "int"
```

Comment:
I like this one too. Although it has an error on first run, we should keep it, just to be sure, the error is cleared when error is fixed.